internal/datacoord/compaction_task_clustering.go

```diff
@@ -295,22 +298,89 @@ func (t *clusteringCompactionTask) processMetaSaved() error {
 
 func (t *clusteringCompactionTask) processStats() error {
 	// just the memory step, if it crashes at this step, the state after recovery is CompactionTaskState_statistic.
+	existNonStats := false
+	tmpToResultSegments := make(map[int64][]int64, len(t.GetTmpSegments()))
```

Comment:
there should be a different behavior depending if stats task is enabled or not

Reply:
wooo! you catch an important point, I will add it.